custom_components/circadian_lighting/switch.py

```diff
@@ -299,7 +300,7 @@ def _calc_rgb(self):
         return color_temperature_to_rgb(self._color_temperature())
 
     def _calc_xy(self):
-        return color_RGB_to_xy(*self._calc_rgb())
+        return list(colour.temperature.CCT_to_xy_CIE_D(self._color_temperature()))
```

Comment:
This is something that's bothered me too but adding additional package requirements has always caused problems in the past for users that have non "standard" installations. If the `colour-science` package fails to install then the entire CL integration will break. I think a better solution would be to wrap this in a try and if an exception occurs to use the current built-in method. I can make that change when I get a chance or feel free to handle it if you're so inclined.

Reply:
Hey, thanks for the feedback. Yep, agree with your point, also because currently the packaged version of `colour-science` has a [problem](https://github.com/colour-science/colour/issues/663) when running on raspberry-pi (this was fixed ~8 months ago on their repo, but it is not yet on the release version). I'll add this check by the end of the week :)

Reply:
If the only method used is `color.temperature.CCT_to_xy_CIE_D()` another option would be to just extract the code for this method (https://github.com/colour-science/colour/blob/a412b670448534987fc25d3a1a09cedc2a3ea5f2/colour/temperature/cie_d.py#L127-L183) from the package and omit the rest of it.

This would completely ditch the dependency while still providing the same functionality.